File: frontend/src/app/features/brands/brand.models.ts
export type BrandStatus = 'ACTIVE' | 'ARCHIVED';

export interface Brand {
  readonly id: string;
  readonly workspaceId: string;
  readonly ownerUserId: string;
  readonly name: string;
  readonly businessType: string | null;
  readonly industry: string | null;
  readonly targetAudience: string | null;
  readonly brandVoice: string | null;
  readonly preferredCta: string | null;
  readonly primaryColor: string | null;
  readonly secondaryColor: string | null;
  readonly website: string | null;
  readonly facebookUrl: string | null;
  readonly instagramUrl: string | null;
  readonly linkedinUrl: string | null;
  readonly tiktokUrl: string | null;
  readonly status: BrandStatus;
  readonly createdAt: string;
  readonly updatedAt: string;
}

export interface CreateBrandPayload {
  readonly name: string;
  readonly businessType: string | null;
  readonly industry: string | null;
  readonly targetAudience: string | null;
  readonly brandVoice: string | null;
  readonly preferredCta: string | null;
  readonly primaryColor: string | null;
  readonly secondaryColor: string | null;
  readonly website: string | null;
  readonly facebookUrl: string | null;
  readonly instagramUrl: string | null;
  readonly linkedinUrl: string | null;
  readonly tiktokUrl: string | null;
}

export interface UpdateBrandPayload extends CreateBrandPayload {
  readonly status: BrandStatus;
}

File: mobile/lib/features/brands/domain/brand_models.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'brand_models.freezed.dart';
part 'brand_models.g.dart';

@freezed
class Brand with _$Brand {
  const factory Brand({
    required String id,
    required String workspaceId,
    required String ownerUserId,
    required String name,
    required String businessType,
    required String industry,
    required String targetAudience,
    required String brandVoice,
    required String preferredCTA,
    String? primaryColor,
    String? secondaryColor,
    String? website,
    String? facebookUrl,
    String? instagramUrl,
    String? linkedinUrl,
    String? tiktokUrl,
    @Default('ACTIVE') String status,
    DateTime? createdAt,
  }) = _Brand;

  factory Brand.fromJson(Map<String, dynamic> json) => _$BrandFromJson(json);
}

@freezed
class CreateBrandRequest with _$CreateBrandRequest {
  const factory CreateBrandRequest({
    required String name,
    required String businessType,
    required String industry,
    required String targetAudience,
    required String brandVoice,
    required String preferredCTA,
    String? primaryColor,
    String? secondaryColor,
    String? website,
    String? facebookUrl,
    String? instagramUrl,
    String? linkedinUrl,
    String? tiktokUrl,
  }) = _CreateBrandRequest;

  factory CreateBrandRequest.fromJson(Map<String, dynamic> json) => _$CreateBrandRequestFromJson(json);
}
